torch/distributed/_functional_collectives.py

```diff
@@ -787,8 +787,23 @@ def reduce_scatter_tensor_inplace(
     return output.copy_(reduce_scatter_tensor(input, op, scatter_dim, group, tag))
 
 
+def all_reduce_inplace(
+    tensor: torch.Tensor,
+    op: str = "sum",
+    group=None,
+    async_op: bool = False,
+    tag: str = "",
+):
+    assert (
```

Comment:
@LucasLLC avoiding async_op was intentional, due to needing dynamo to understand it and model it if we were going to produce one and return it from the function.